test/config-luatest/names_upgrade_test.lua

```diff
@@ -101,31 +100,59 @@ local function assert_before_upgrade()
     t.assert_equals(info.replicaset.name, nil)
 end
 
-local function assert_after_upgrade(instance_name, replicaset_name, names)
+local function wait_names_after_upgrade(instance_name, replicaset_name, names)
     t.helpers.retrying({timeout = 20}, function()
-        local info = box.info
-        t.assert_equals(info.name, instance_name)
-        t.assert_equals(info.replicaset.name, replicaset_name)
-
-        local alerts = require('config'):info().alerts
-        for name, _ in pairs(names) do
-            t.assert_equals(alerts[name], nil)
-        end
+        _G.assert_names_after_upgrade(instance_name, replicaset_name, names)
     end)
 end
 
 g.test_upgrade = function()
-    g.instance_1:exec(assert_before_upgrade)
-    g.instance_2:exec(assert_before_upgrade)
+    g.instance_1:exec(assert_names_before_upgrade)
+    g.instance_2:exec(assert_names_before_upgrade)
 
     g.instance_1:exec(function()
         box.ctl.wait_rw()
-        box.schema.upgrade()
+        t.assert_equals(box.space._schema:get{'version'}, {'version', 2, 11, 0})
+        -- Pretend, that we upgrade to 2.11.5 version, since it's
+        -- the first 2.11 release, which properly works with names.
+        box.internal.run_schema_upgrade(function()
+            box.space._schema:delete("max_id")
+            box.space._schema:replace{'version', 2, 11, 5}
+        end)
```

Comment:
I don't want to be picky about test cases, but there are some thoughts around.

In the previous version of the testing scenario we verify a user journey: upgrading from 2.11.0 to the latest version.

The relaxation of the schema requirements that is implemented in this patch opens the second user scenario: running a non-upgraded 3.x. instance on the 2.11.5 schema version.

The test case now verifies a partial upgrade scenario (from 2.11.0 to 2.11.5), but it is not accessible for a user. However, two *user* scenarios above are not directly verified.

Should we write separate test cases for upgrading 2.11.0 to latest and for starting on 2.11.5?

Reply:
Done. Do we want the same for credentials? I don't see much sense in testing upgrade from 2.11.5 in case of names and from 2.11.1 in case of credentials, since it's already covered by upgrade from 2.11.0, in which we go through the intermediate step of upgrading to 2.11.1 and 2.11.5

Reply:
I'm a bit worrying about such 'stop upgrading' steps that use knowledge about internals. It feels fragile.

I'm OK, but I'll reach you (in the good sense) if something goes wrong with it :)

Taking into account that the upgrading tests are not easy to write, I think it is nice that we have some.